legend-engine-xts-relationalStore/legend-engine-xt-relationalStore-generation/legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/modelToModelToRelational/milestoned_graphFetch_m2m2r.pure

```diff
@@ -0,0 +1,212 @@
+                                                      owner = $mapping)
+                          );
+}
+
+function <<test.Test, test.AlloyOnly>> meta::pure::graphFetch::tests::m2m2r::milestoning::toNonMilestonedTargetProperty::testToNonMilestonedTargetProperty_AllVersions():Boolean[1]
+{
+  let mapping =meta::relational::tests::m2m2r::milestoning::toNonMilestonedTargetProperty::TargetToModelMappingWithMilestonedComplexPropertyFlattened_AllVersions;
+  let runtime=  getModelChainRuntime($mapping);
+
+   let query={bdate:DateTime[1]|meta::relational::tests::milestoning::TargetProductFlattened.all($bdate)->graphFetch(
+      #{
+        meta::relational::tests::milestoning::TargetProductFlattened{
+          id,
+          name,
+          synonymNames
+        }
+      }#
+    )->serialize(
+      #{
+        meta::relational::tests::milestoning::TargetProductFlattened{
+          id,
+          name,
+          synonymNames
+        }
+      }#
+    )
+    ->meta::pure::mapping::from($mapping,$runtime)
+    };
+   let result = meta::legend::executeLegendQuery($query, [pair('bdate', '2023-10-15T00:00:00')], ^meta::pure::runtime::ExecutionContext(), meta::relational::extension::relationalExtensions());
+   assertJsonStringsEqual('{"builder":{"_type":"json"},"values":[{"id":2,"name":"ProductName2","synonymNames":["GS-Mod-S0","GS-Mod-S1","GS-Mod-S2","GS-Mod-S3"]},{"id":3,"name":"ProductName3","synonymNames":["GS-Mod-S2","GS-Mod-S3"]}]}', $result);
+}
+
+function meta::pure::graphFetch::tests::m2m2r::milestoning::getModelChainRuntime(m2m:meta::pure::mapping::Mapping[1]):Runtime[1]
+{
+ ^meta::pure::runtime::Runtime(
+      connections=[
+        meta::pure::graphFetch::tests::milestoning::getAlloyTestH2Connection(),
+        ^meta::pure::mapping::modelToModel::ModelChainConnection(element = ^meta::pure::mapping::modelToModel::ModelStore(), mappings = [$m2m])
+      ]
+  );
+}
+
+Class <<temporal.businesstemporal>> meta::relational::tests::milestoning::TargetProduct1
+{
+  id: Integer[1];
+  name: String[1];
+  type: String[1];
+  synonyms: meta::relational::tests::milestoning::TargetProductSynonym[*];
+}
+
+
+Class <<temporal.businesstemporal>> meta::relational::tests::milestoning::TargetProductSynonym
+{
+  synonym: String[1];
+}
+
+Class <<temporal.businesstemporal>> meta::relational::tests::milestoning::TargetProductFlattened
+{
+  id: Integer[1];
+  name: String[1];
+  type: String[1];
+  synonymNames: String[*];
+}
+
+###Mapping
+Mapping meta::relational::tests::m2m2r::milestoning::toMilestonedTargetProperty::TargetToModelMappingWithMilestonedComplexProperty
+(
+  include meta::relational::tests::milestoning::milestoningmap
+
+  *meta::relational::tests::milestoning::TargetProduct1: Pure
+  {
+    ~src meta::relational::tests::milestoning::Product
+    id: $src.id,
+    name: $src.name,
+    synonymsAllVersions: $src.synonymsAllVersions
```

Comment:
I wonder if there are better ways to do this?  Seems confusing :(

Reply:
should be  synonyms: $src.synonyms ?

Reply:
Yeah, I wonder if from the Studio grammar perspective this is more understandable to users.  We can internally convert to 'AllVersions" if requried.

Reply:
Yes that seems more intuitive . we can support both patterns. it needs a fix in m2m source tree calculation